### PR TITLE
gitea: update to 1.12.2, fix cross build

### PR DESCRIPTION
**Pull request by fosslinux**
The upstream tarball has been changed to gitea's hand-generated ones
because of the recent sha256sum debacle, as well as their existence
(they should be preferred). In addition, create_wrksrc=yes is now
required.

gitea also now uses nodejs to generate css and js files. However, in the
tarballs, they pre-generate them. We use this.

Configuration file changes have been moved to a .patch because the changes
were getting very dense and difficult to read in a sed.

There was a strange error with go modules which was meant to be fixed by
an upstream patch. 12030.patch is this patch. It doesn't work completely
but is fully fixed by running `go mod vendor` in pre_build (NB: the
patch is still needed). This may not be nessecary for the next release.

@the-maldridge

### CODE REVIEW

File: srcpkgs/gitea/patches/ldflags.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- Makefile	2020-07-10 12:25:58.893000000 +1000
++++ Makefile	2020-07-10 12:26:04.329000000 +1000
+@@ -84,7 +84,7 @@
+ 	endif
+ endif
+ 
+-LDFLAGS := $(LDFLAGS) -X "main.MakeVersion=$(MAKE_VERSION)" -X "main.Version=$(GITEA_VERSION)" -X "main.Tags=$(TAGS)"
++LDFLAGS := $(GOLDFLAGS) -X "main.MakeVersion=$(MAKE_VERSION)" -X "main.Version=$(GITEA_VERSION)" -X "main.Tags=$(TAGS)"
```
Comment (ericonr) [inline srcpkgs/gitea/patches/ldflags.patch:8]: Isn't it easier to unset LDFLAGS when running `make`?

I guess the build failed when you didn't have this patch?

Comment (ericonr) [reply srcpkgs/gitea/patches/ldflags.patch:8]: LDFLAGS are being used for cgo afaiu, which can indeed use them.

Comment (fosslinux) [reply srcpkgs/gitea/patches/ldflags.patch]: Patch removed.

**Comment by fosslinux** [comment]
Ok, sorry about that very bad first implementation where I wasted my and everyone else's time trying to get a stupid idea to work. :( I will think more carefully in future about why something is a certain way....

This is the same thing just simpler and with consideration and thinking and without stupidity.

Builds on all archs and runs on x86_64 x86_64-musl and aarch64 (tested)

File: srcpkgs/gitea/template [hunk 2]
```diff
@@ -42,31 +42,12 @@ make_dirs="/var/lib/gitea 0755 _gitea _gitea
...
-		/TEMP_PATH/s:data/tmp/uploads:/var/lib/gitea/tmp/uploads:
-		/LFS_CONTENT/s:data/lfs:/var/lib/gitea/lfs:
-		s:data/gitea.db:/var/lib/gitea/&:
-		s:indexers/issues.bleve:/var/lib/gitea/&:
-		s:data/avatars:/var/lib/gitea/&:
-		s:SSH_ROOT_PATH =:& /var/lib/gitea:
-		s:STATIC_ROOT_PATH =:& /var/lib/gitea:
-		s:^ROOT_PATH =:& /var/log/gitea:' custom/conf/app.ini.sample
+	go mod vendor
+	CGO_CFLAGS= CGO_LDFLAGS= make generate
```
Comment (the-maldridge) [inline srcpkgs/gitea/template:46]: if you're using the provided tarballs, this step should no longer be needed.

Comment (fosslinux) [reply srcpkgs/gitea/template]: You are correct, checked.

**Comment by fosslinux** [comment]
@the-maldridge Can you look at this again when you get a minute

**Pull request merged**